eodatasets3/properties.py

```diff
@@ -234,6 +234,7 @@ class StacPropertyView(collections.abc.Mapping):
         "odc:producer": producer_check,
         "odc:product_family": None,
         "odc:region_code": None,
+        "odc:dataset_maturity": None,
```

Comment:
As mentioned in the previous comment -- this was an existing bug (not introduced by you), and we should use `dea:dataset_maturity` instead of adding this one.

Reply:
Nice spotting.  Fixed.